src/components/batch-models-multi-select/CopyInUserBatchModelsMultiSelect.component.js

```diff
@@ -0,0 +1,224 @@
+                })
+            )
+            .catch(err =>
+                this.close(this.getTranslation("error_loading_data") + " :" + err.toString())
+            );
+    }
+
+    close(snackMessage = null) {
+        if (snackMessage) snackActions.show({ message: snackMessage });
+        this.props.onRequestClose();
+    }
+
+    renderStrategyToggle() {
+        if (this.state.parents && this.state.parents.length > 1) {
+            const label =
+                this.getTranslation("update_strategy") +
+                ": " +
+                this.getTranslation("update_strategy_" + this.state.updateStrategy);
+
+            return (
+                <Toggle
+                    label={label}
+                    style={this.styles.toggle}
+                    checked={this.state.updateStrategy === "replace"}
+                    onToggle={(ev, newValue) =>
+                        this.setState({ updateStrategy: newValue ? "replace" : "merge" })
+                    }
+                />
+            );
+        } else {
+            return null;
+        }
+    }
+    async copyInUserSave() {
+        const { parents, selectedIds, copyUserGroups, copyUserRoles } = this.state;
+        await this.props.model
+            .copyInUserSave(parents, selectedIds, copyUserGroups, copyUserRoles)
+            .then(() => this.close(this.props.onSuccess))
+            .catch(err => this.close(this.props.onError));
+    }
+    onChange(selectedIds) {
+        this.setState({ selectedIds });
+    }
+    render() {
+        switch (this.state.state) {
+            case "loading":
+                return (
+                    <div style={this.styles.loadingMask}>
+                        <LoadingMask />
+                    </div>
+                );
+            case "error":
+                return <div style={this.styles.loadingMask}>{this.state.error}</div>;
+            case "ready":
+                return this.renderForm();
+            default:
+                throw new Error(`Unknown state: ${state}`);
+        }
+    }
+
+    onFilterTextChange(event) {
+        this.setState({ filterText: event.target.value });
+    }
+    getDialogButtons() {
+        return [
+            <FlatButton
+                label={this.getTranslation("cancel")}
+                onClick={this.props.onRequestClose}
+                style={this.styles.cancelButton}
+            />,
+            <RaisedButton
+                primary
+                label={this.getTranslation("save")}
+                onClick={ this.state.copyUserGroups || this.state.copyUserRoles ? this.copyInUserSave.bind(this) : err => this.close("Error: You must select a toggle option") }
+            />,
+        ];
+    }
+
+    render() {
+        const isLoading = this.state.state === "loading";
+        const { parents, allChildren, filterText, selectedIds } = this.state;
+        const title = this.props.getTitle(parents, allChildren);
+        const options = _(allChildren || [])
```

Comment:
I was thinking: should we remove the user being copied from the available destinations? it does not make sense to copy onto itself, I guess.